### ISSUE TITLE
Integrate ticker channels into structured concurrency and flows

### ISSUE THREAD

**Issue by elizarov** [Kotlin/kotlinx.coroutines#540]
We need to integrate ticker channels into a structured concurrency story to make sure they are not lost running forever. They should be somehow marked as "daemon" (?) children of their parent and should not prevent the parent completion, but shall be cancelled when all other (non-daemon?) children coroutines complete. This would make standard timing-related operations easy to write in a less error-prone way.

P.S. This is not critical for 1.0 release, since the API for channels is going to be kept experimental anyway.

**Comment by wyaeld** [comment]
@elizarov is this coming? Currently trying to build coroutine-based background processing but having a time-keeping channel will simplify a number of things.

**Comment by elizarov** [comment]
@wyaeld Yes. Ticker channels will become a part of `Flow` framework. Yet, this is not going to happen in the short-term future. Actually, flows makes it so easy to write this kind of things yourself, as to almost remove the need for any kind of library primitive. 

Can you, please, write more details on what kind of background processing you are doing and what are your use-cases for time-keeping channels.

**Comment by wyaeld** [comment]
Sure.  I'm reading through all the Flow docs this week to understand how
that works, since that was my assumption on how you would do it.
In my case I'm writing at IoT control element, which is continually
processing commands that get delivered in batches, and for which I want an
observer/reporting type element that is able to report say every 10s, on
the status of everything in flight.  This command batches can be hundreds
of individual messages that have to go out.  The TickerChannel seemed like
a useful trigger for the reporting element.

**Comment by elizarov** [comment]
@wyaeld For your use-case we plan something like a specialized time-based `chunked` primitive. See #1302

**Comment by esdudnik** [comment]
Will ticker channels be multiplatform? I came from this issue
https://github.com/Kotlin/kotlinx.coroutines/issues/1186
Looking for something similar for multiplatform library.

**Comment by elizarov** [comment]
Yes. The plan is to have them flow-based an multiplatform.

**Comment by MarcelReiter** [comment]
Are there any updates?

We'd like to use this feature in an app displaying a one-time-password for two-step-verification similar to the [google authenticator](https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2)

Usually, OTPs are valid 30 seconds, and "start" on a full minute or on a half minute,
e.g. 12:00:00 or 12:00:30, but not 12:00:01

In the moment we're using rx-java and emit:
- on creation
- after <first delay>, which is the time until the next 30s-interval; e.g. when it's 12:00:03 then the first delay is 27s
- each 30s after that

**Comment by YoshiRulz** [comment]
If you want another use-case: I'm writing a daemon which periodically checks the audio graph of an (external) [PipeWire](https://pipewire.org) daemon, and creates or deletes links based on a user-defined ruleset. At the moment I'm struggling to get *any* timer to work, including the deprecated `ticker`. Here's my code:
```kt
public fun main(): Unit = runBlocking {
	JVMSoundController().use { soundController ->
		launch {
			embeddedServer(Netty, port = wsProtocolPort) {
				// ... (the other half of my app, a WebSocket server for remote control)
			}.start(wait = true)
		}
		launch {
			flow {
				while (true) emit(Unit)
			}.onEach { delay(3.seconds) }.collect {
				println("A")
//				soundController.checkLocks()
			}
		}
	}
}
```

**Comment by hugo891** [comment]
> We need to integrate ticker channels into a structured concurrency story to make sure they are not lost running forever. They should be somehow marked as "daemon" (?) children of their parent and should not prevent the parent completion, but shall be cancelled when all other (non-daemon?) children coroutines complete. This would make standard timing-related operations easy to write in a less error-prone way.
> 
> P.S. This is not critical for 1.0 release, since the API for channels is going to be kept experimental anyway.

https://github.com/Kotlin/kotlinx.coroutines/issues/540#issue-358386966

**Comment by akuter-bc** [comment]
Hi guys.
Is this still planned on a future release?
